sky/provision/oci/instance.py

```diff
@@ -306,12 +310,12 @@ def cleanup_ports(
     provider_config: Optional[Dict[str, Any]] = None,
 ) -> None:
     """Delete any opened ports."""
-    del cluster_name_on_cloud, ports, provider_config
-    # OCI ports in security groups are opened while creating the new
-    # VCN (skypilot_vcn). The VCN will only be created at the first
-    # time when it is not existed. We'll not automatically delete the
-    # VCN while teardown clusters. it is intended to let user to decide
-    # to delete the VCN or not from OCI console, for example.
+    assert provider_config is not None, cluster_name_on_cloud
+    region = provider_config['region']
+    query_helper.remove_ingress_rules(
+        region=region,
+        cluster_name=cluster_name_on_cloud,
+        ports=resources_utils.port_ranges_to_set(ports))
```

Comment:
Should we directly cleanup the whole security groups?

Reply:
Yes! My implementation considerred the case where incrementally add/delete ports or user manually editted the nsg to add their ports. It is simpler if we can remove it as a whole. I'll change it to deal the nsg as a whole.

Reply:
Modified.